pageserver/src/tenant/timeline.rs

```diff
@@ -3735,6 +3784,7 @@ impl Timeline {
             guard
                 .try_freeze_in_memory_layer(at, &self.last_freeze_at, write_lock)
                 .await
+                .expect("layer manager cannot be shutdown, because we are holding the write_lock")
```

Comment:
I failed to clean this up just by itself, instead of trimming down on "fix: unify freeze+signal" which unifies the jungle of different methods for freezing, signalling flush after freeze, signalling flush after non-freeze, and maybe waiting for the flush loop to reach requested count.